// js-pert/build/completionProbability.test.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const completionProbability_1 = __importDefault(require("./completionProbability"));
const round_1 = __importDefault(require("./round"));
const network_1 = require("./network");
const network_test_1 = require("./network.test");
const activities_test_1 = require("./activities.test");
const criticalPath_test_1 = require("./criticalPath.test");
const pert = {
    activitiesParams: activities_test_1.activitiesParams,
    network: network_test_1.network,
    earliestFinishTimes: {},
    latestStartTimes: {},
    earliestStartTimes: {},
    latestFinishTimes: { [network_1.END]: 20 },
    slack: {},
    criticalPath: criticalPath_test_1.criticalPath,
};
test('get completion probability on half of normal distribution', () => {
    expect(round_1.default(completionProbability_1.default(pert, 20), 1)).toBe(0.5);
});
test('get completion probability', () => {
    expect(round_1.default(completionProbability_1.default(pert, 15), 6)).toBe(0.011084);
});
