// js-pert/build/pert.test.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const pert_1 = __importDefault(require("./pert"));
const network_test_1 = require("./network.test");
const activities_test_1 = require("./activities.test");
const criticalPath_test_1 = require("./criticalPath.test");
const times_test_1 = require("./times.test");
test('generate pert serie', () => {
    expect(pert_1.default(network_test_1.activities)).toEqual({
        activitiesParams: activities_test_1.activitiesParams,
        network: network_test_1.network,
        earliestFinishTimes: times_test_1.earliestFinishTimes,
        latestStartTimes: times_test_1.latestStartTimes,
        earliestStartTimes: {
            __start: 0,
            A: 0,
            D: 6,
            I: 8,
            B: 0,
            E: 3,
            C: 0,
            F: 4,
            H: 9,
            G: 4,
            J: 14,
            __end: 17,
        },
        latestFinishTimes: {
            __end: 17,
            I: 17,
            D: 12,
            A: 10,
            J: 17,
            H: 14,
            E: 9,
            B: 6,
            F: 9,
            G: 14,
            C: 4,
            __start: 0,
        },
        slack: {
            __end: 0,
            I: 4,
            D: 4,
            A: 4,
            J: 0,
            H: 0,
            E: 3,
            B: 3,
            F: 0,
            G: 7,
            C: 0,
            __start: 0,
        },
        criticalPath: criticalPath_test_1.criticalPath,
    });
});
